archive/brotli.h

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2024 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef ARCHIVE_BROTLI_H_
+#define ARCHIVE_BROTLI_H_
+
+#include <tl/expected.hpp>
+
+#include <cstddef>
+#include <cstdint>
+#include <span>
+#include <string_view>
+#include <vector>
+
+namespace archive {
+
+enum class BrotliError : std::uint8_t {
```

Comment:
This name is kinda stepping on Brotli's namespace isn't it? Can we use ArchiveError here so it could be reused?

That's an actual question, I'm not sure what the desired level of abstraction around 3rd party libs should be?

Reply:
Brotli's in C and this lives in a C++-namespace, so it should be fine. We'll probably want a unified interface towards the `//archive` stuff at some point, but for now each library has its own set of errors with nothing holding them together other than all of them having the same `expected<vector<byte>, LibError> lib_decode(span<byte const>)`-API.

Reply:
Once this is merged, we'll be good on content encoding; I'm not sure we need to bother with lzw at all. But I think this is a concern for *after* this PR is merged, not before. Can add an issue to the tracker for it and assign it to me if you want.

Reply:
lzw is super rare when it comes to content encodings, but I'm messing around with it in other contexts, so it'll be ~free for us to support if we need it elsewhere anyway.

I think it's better to hold off on a unified archive API for now. I'm messing around with making the engine bits a bit more modular so that users of it can add their own or opt out of the default stuff that we have right now, e.g. for experimenting with adding more image encodings or transfer encodings via some form of plugin-API. Might change how we use the archive bits, so tying all archive-implementations together before we see what happens there doesn't feel worthwhile.